src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBioMartFileProcessor.java

```diff
@@ -0,0 +1,167 @@
+             |              |
+             |              +---ENST000020
+             |                      |
+             |                      + ILMN_121314, ILMN_141617,...
+             |
+             +mmusculus_proteinsToGenes
+             |              |
+             |              +---ENSMUSP000020
+             |              |       |
+             |              |       + ENSMUSG00009, ENSMUSG00010,...
+             |              |
+             |              +---ENSMUSP000021
+             |                      |
+             |                      + ENSMUSG00011, ENSMUSG00012,...
+
+         */
+
+        Map<String, Map<String, List<String>>> mappings = new HashMap<>();
+        // Read each file in BioMart directory and iterate through each line.
+        try (Stream<Path> paths = Files.walk(this.pathToFile)) {
+            paths
+                .filter(Files::isRegularFile)
+                .forEach(bioMartFile -> {
+                    // Get species name from filename
+                    String biomartFileSpecies = bioMartFile.getFileName().toString().split("_")[0];
+                    // Mappings that will be generated from each file for the species.
+                    Map<String, List<String>> proteinToTranscripts = new HashMap<>();
+                    Map<String, List<String>> proteinToGenes = new HashMap<>();
+                    Map<String, List<String>> transcriptToMicroarrays = new HashMap<>();
+                    Map<String, List<String>> uniprotToProteins = new HashMap<>();
+                    try {
+                        for (String biomartLine : Files.readAllLines(bioMartFile)) {
+                            // Split each tab-separated line. Each line has four values. The first three are always
+                            // Ensembl Gene (ENSG), Transcript (ENST) and Protein (ENSP), in that order.
+                            // The 4th can be a UniProt or microarray identifier, depending on the file being read.
+                            // None of the values are guaranteed to in the line, except perhaps for the first (Gene).
+                            List<String> tabSplit = Arrays.asList(biomartLine.split("\t"));
+                            // Processing of UniProt mapping files.
+                            // UniProt identifier mapping files are used to fully populate
+                            // the 'uniprotToProteins' and 'proteinToGenes' mappings.
+                            if (bioMartFile.toString().endsWith("uniprot")) {
+                                // 'uniprotToProteins' mapping requires the 3rd (protein) and 4th (uniprot) values in the line.
+                                if (properArraySizeAndNecessaryColumnsContainData(tabSplit, 3,proteinIdentifierIndex,identifierIndex)) {
+                                    uniprotToProteins = mapIdentifiers(uniprotToProteins, tabSplit.get(identifierIndex), tabSplit.get(proteinIdentifierIndex));
```

Comment:
If this method takes the file lines and the indexes for the columns for the key and value to include in the map, the map can be populated as in one object that is returned when the population is complete:

```
uniprotToProteins = getIdentifiersMapping(bioMartFilesLines, identifierIndex, proteinIdentifierIndex);

private Map<String, List<String>> getIdentifiersMapping(List<String> bioMartFileLines, int keyIndex, int valueIndex) {
    Map<String, List<String>> identifierMapping = new HashMap<>();
    
    for (String bioMartFileLine : bioMartFileLines) {
        List<String> tabSplit = Arrays.asList(bioMartFileLine.split("\t"));
        String mapKey = tabSplit.get(keyIndex);
        String mapValue = tabSplit.get(valueIndex);
        
        identifierMapping.computeIfAbsent(mapKey, k -> new ArrayList<>()).add(mapValue);
        return identifierMapping;
    }
    
    return identifierMapping;
}
```